File: src/gatling/scala/org/openapitools/client/model/QueryComment.scala

package org.openapitools.client.model


case class QueryComment (
    /* Sequence number of the comment/action in the query history */
    _sequence: Option[Integer],
    /* Status of the query after this comment (e.g., Open, Closed) */
    _annotationStatus: Option[String],
    /* Username of the user who made the comment or action */
    _user: Option[String],
    /* Text of the comment */
    _comment: Option[String],
    /* Whether the query was marked closed at this step */
    _closed: Option[Boolean],
    /* Date of the comment or action */
    _date: Option[String]
)
object QueryComment {
    def toStringBody(var_sequence: Object, var_annotationStatus: Object, var_user: Object, var_comment: Object, var_closed: Object, var_date: Object) =
        s"""
        | {
        | "sequence":$var_sequence,"annotationStatus":$var_annotationStatus,"user":$var_user,"comment":$var_comment,"closed":$var_closed,"date":$var_date
        | }
        """.stripMargin
}

File: client/zapier/models/QueryComment.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}sequence`,
                label: `Sequence number of the comment/action in the query history - [${labelPrefix}sequence]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}annotationStatus`,
                label: `Status of the query after this comment (e.g., Open, Closed) - [${labelPrefix}annotationStatus]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}user`,
                label: `Username of the user who made the comment or action - [${labelPrefix}user]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}comment`,
                label: `Text of the comment - [${labelPrefix}comment]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}closed`,
                label: `Whether the query was marked closed at this step - [${labelPrefix}closed]`,
                type: 'boolean',
            },
            {
                key: `${keyPrefix}date`,
                label: `Date of the comment or action - [${labelPrefix}date]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'sequence': bundle.inputData?.[`${keyPrefix}sequence`],
            'annotationStatus': bundle.inputData?.[`${keyPrefix}annotationStatus`],
            'user': bundle.inputData?.[`${keyPrefix}user`],
            'comment': bundle.inputData?.[`${keyPrefix}comment`],
            'closed': bundle.inputData?.[`${keyPrefix}closed`],
            'date': bundle.inputData?.[`${keyPrefix}date`],
        }
    },
}
